### PR TITLE
Storybook updates

### LINKED ISSUE #147: Storybook as repo documentation
## What problem does this address?
Actually we haven't a (easy to reach) user documentation

## What is your proposed solution?
Using storybook we can create a little nicer documentation than the traditional one, and also we can show the components of the project and play with them. Using `build-storybook -o docs/`, should published and shown in the right panel of the Github repo home page

Here an example:
[feature-notifications Storybook](https://dev.codekraft.it/navtest/wp-notify/?path=/story/feature-notifications-introduction--page)

### LINKED ISSUE #241: Conflicting peer dependencies on npm install
### Description

npm install brings multiple errors around peer dependencies.

It may be that this is a conflict between certain WP dependencies and Storybook.

### Step-by-step reproduction instructions

1. Run npm install
2. Check the console for errors

### Screenshots, screen recording, code snippet

<details>
<summary>Expand for details</summary>
<code>
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-element-to-jsx-string@14.3.4
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN node_modules/react-element-to-jsx-string
npm WARN   react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.16
npm WARN   node_modules/@storybook/react
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN   node_modules/react-element-to-jsx-string
npm WARN     react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.16
npm WARN     node_modules/@storybook/react
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-element-to-jsx-string@14.3.4
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">= 16.8.6" from @design-systems/utils@2.12.0
npm WARN   node_modules/@devtools-ds/themes/node_modules/@design-systems/utils
npm WARN     @design-systems/utils@"2.12.0" from @devtools-ds/themes@1.2.0
npm WARN     node_modules/@devtools-ds/themes
npm WARN   145 more (@floating-ui/react-dom, @storybook/addon-actions, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN node_modules/react-element-to-jsx-string
npm WARN   react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.16
npm WARN   node_modules/@storybook/react
npm WARN 
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1" from react-element-to-jsx-string@14.3.4
npm WARN   node_modules/react-element-to-jsx-string
npm WARN     react-element-to-jsx-string@"^14.3.4" from @storybook/react@6.5.16
npm WARN     node_modules/@storybook/react
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-inspector@5.1.1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN node_modules/react-inspector
npm WARN   react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.16
npm WARN   node_modules/@storybook/addon-actions
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN   node_modules/react-inspector
npm WARN     react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.16
npm WARN     node_modules/@storybook/addon-actions
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit@1.3.11
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.0 || ^17.0.0" from reakit@1.3.11
npm WARN node_modules/reakit
npm WARN   reakit@"^1.3.8" from @wordpress/components@23.5.0
npm WARN   node_modules/@wordpress/components
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.0 || ^17.0.0" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN     reakit@"^1.3.8" from @wordpress/components@23.5.0
npm WARN     node_modules/@wordpress/components
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit@1.3.11
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">= 16.8.6" from @design-systems/utils@2.12.0
npm WARN   node_modules/@devtools-ds/themes/node_modules/@design-systems/utils
npm WARN     @design-systems/utils@"2.12.0" from @devtools-ds/themes@1.2.0
npm WARN     node_modules/@devtools-ds/themes
npm WARN   145 more (@floating-ui/react-dom, @storybook/addon-actions, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^16.8.0 || ^17.0.0" from reakit@1.3.11
npm WARN node_modules/reakit
npm WARN   reakit@"^1.3.8" from @wordpress/components@23.5.0
npm WARN   node_modules/@wordpress/components
npm WARN 
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^16.8.0 || ^17.0.0" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN     reakit@"^1.3.8" from @wordpress/components@23.5.0
npm WARN     node_modules/@wordpress/components
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit-system@0.15.2
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.0 || ^17.0.0" from reakit-system@0.15.2
npm WARN node_modules/reakit-system
npm WARN   reakit-system@"^0.15.2" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.0 || ^17.0.0" from reakit-system@0.15.2
npm WARN   node_modules/reakit-system
npm WARN     reakit-system@"^0.15.2" from reakit@1.3.11
npm WARN     node_modules/reakit
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit-system@0.15.2
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">= 16.8.6" from @design-systems/utils@2.12.0
npm WARN   node_modules/@devtools-ds/themes/node_modules/@design-systems/utils
npm WARN     @design-systems/utils@"2.12.0" from @devtools-ds/themes@1.2.0
npm WARN     node_modules/@devtools-ds/themes
npm WARN   145 more (@floating-ui/react-dom, @storybook/addon-actions, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^16.8.0 || ^17.0.0" from reakit-system@0.15.2
npm WARN node_modules/reakit-system
npm WARN   reakit-system@"^0.15.2" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN 
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^16.8.0 || ^17.0.0" from reakit-system@0.15.2
npm WARN   node_modules/reakit-system
npm WARN     reakit-system@"^0.15.2" from reakit@1.3.11
npm WARN     node_modules/reakit
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit-utils@0.15.2
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.0 || ^17.0.0" from reakit-utils@0.15.2
npm WARN node_modules/reakit-utils
npm WARN   reakit-utils@"^0.15.2" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN   2 more (reakit-system, reakit-warning)
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.0 || ^17.0.0" from reakit-utils@0.15.2
npm WARN   node_modules/reakit-utils
npm WARN     reakit-utils@"^0.15.2" from reakit@1.3.11
npm WARN     node_modules/reakit
npm WARN     2 more (reakit-system, reakit-warning)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit-utils@0.15.2
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">= 16.8.6" from @design-systems/utils@2.12.0
npm WARN   node_modules/@devtools-ds/themes/node_modules/@design-systems/utils
npm WARN     @design-systems/utils@"2.12.0" from @devtools-ds/themes@1.2.0
npm WARN     node_modules/@devtools-ds/themes
npm WARN   145 more (@floating-ui/react-dom, @storybook/addon-actions, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^16.8.0 || ^17.0.0" from reakit-utils@0.15.2
npm WARN node_modules/reakit-utils
npm WARN   reakit-utils@"^0.15.2" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN   2 more (reakit-system, reakit-warning)
npm WARN 
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^16.8.0 || ^17.0.0" from reakit-utils@0.15.2
npm WARN   node_modules/reakit-utils
npm WARN     reakit-utils@"^0.15.2" from reakit@1.3.11
npm WARN     node_modules/reakit
npm WARN     2 more (reakit-system, reakit-warning)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: reakit-warning@0.6.2
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.0 || ^17.0.0" from reakit-warning@0.6.2
npm WARN node_modules/reakit-warning
npm WARN   reakit-warning@"^0.6.2" from reakit@1.3.11
npm WARN   node_modules/reakit
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.0 || ^17.0.0" from reakit-warning@0.6.2
npm WARN   node_modules/reakit-warning
npm WARN     reakit-warning@"^0.6.2" from reakit@1.3.11
npm WARN     node_modules/reakit
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @mdx-js/react@1.6.22
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peer react@">= 16.8.6" from @devtools-ds/object-inspector@1.2.0
npm WARN   node_modules/@devtools-ds/object-inspector
npm WARN     @devtools-ds/object-inspector@"^1.1.2" from @storybook/addon-interactions@6.5.15
npm WARN     node_modules/@storybook/addon-interactions
npm WARN   160 more (@devtools-ds/themes, @design-systems/utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN node_modules/@storybook/addon-docs/node_modules/@mdx-js/react
npm WARN   @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.16
npm WARN   node_modules/@storybook/addon-docs
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN   node_modules/@storybook/addon-docs/node_modules/@mdx-js/react
npm WARN     @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.16
npm WARN     node_modules/@storybook/addon-docs
npm WARN deprecated @types/vfile-message@2.0.0: This is a stub types definition. vfile-message provides its own type definitions, so you do not need this installed.
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
</code>
</details>

### Environment info

Local install outside of a WordPress environment.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except WP Feature Notifications.

Yes

---
Will be closed with: #236

### PR DESCRIPTION
**Pull request by erikyo**
## What?
#147 Add an autogenerated documentation to Github pages

## Why?
Actually we haven't a (easy to reach) user documentation. 

## How?
Thanks to the  capability of storybook to generate a static website we have the possibility to see the components in action and have a mini-site to show the documentation (while still using md files, if we prefer, as I did [here](https://raw.githubusercontent.com/erikyo/wp-feature-notifications/docs/storybook/src/stories/docs/restApi.stories.mdx))

close #147, and closes #241

I had to synchronize [the branch with the master](https://github.com/WordPress/wp-feature-notifications/pull/148/commits/49aeff77f7ea0dfb8b8e18aea0b60871e87649c1), which also forced me to update some dependencies. Little harm either way, maybe we get fewer alerts from dependbot

### CODE REVIEW

File: docs/database-schema.md
Comment (johnhooks) [inline docs/database-schema.md:13]: @erikyo why is the docs folder being deleted?

Comment (erikyo) [inline docs/database-schema.md]: i'm reverting that changes... it's due a failed merge! sorry

File: package.json
Comment (johnhooks) [inline package.json]: Can we keep these in the root `package.json` even if you need them elsewhere.

File: package.json
Comment (johnhooks) [inline package.json]: Can `@types/jest` be left in `devDependencies`?

File: src/storybook/stories/docs/database-schema.md
Comment (johnhooks) [inline src/storybook/stories/docs/database-schema.md]: @erikyo this hasn't been merged into `develop` yet, you should probably add the original schema doc back.

Comment (erikyo) [reply src/storybook/stories/docs/database-schema.md]: yep i know (seems the only thing that isn't merged correctly at the moment), this is due the pr was created time ago and wasn't synced properly!

**Review by johnhooks — COMMENTED** [review]
LGTM

**Comment by erikyo** [comment]
^^ updating deps wasn't something that I didn't want to do in this pr, but I was forced

**Review by Sephsekla — DISMISSED** [review]
This is great, it's awesome to have such nice docs!

**Comment by erikyo** [comment]
TODO: after creating the master branch we need to automate this documentation build process with the ci and run it at every merge between development and master

File: .gitignore
Comment (johnhooks) [inline .gitignore]: @erikyo did you mean to add a line here?

**Comment by johnhooks** [comment]
@erikyo I don't like the idea of all the storybook build artifacts being checked into the repo. Is the `docs` for publishing on GitHub pages?

**Comment by johnhooks** [comment]
Can we add a CI workflow for publishing to GitHub pages?

File: .gitignore
Comment (erikyo) [reply .gitignore]: no... however it goes, let's say a collateral damage of the lint. I would leave it since, theoretically, the empty end line is required at the end of each document

**Comment by erikyo** [comment]
> Can we add a CI workflow for publishing to GitHub pages?

I have already created an [issue ](https://github.com/WordPress/wp-feature-notifications/pull/148#issuecomment-1485219735)for this, however I would do it in a separate pr. if you think we'll take care of it shortly I'll remove what we have now in the docs folder, otherwise maybe we'd better leave it there

File: .gitignore
Comment (johnhooks) [reply .gitignore]: Makes sense to me, I agree empty lines between groups is standard.

**Comment by johnhooks** [comment]
I'll help you resolve it now. If we check those in, I think we will regret it quickly.

**Comment by erikyo** [comment]
This is the CI workflow used alongside the Gutenberg repository
https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/storybook-pages.yml

Seems quite easy to add, what o you think @johnhooks?

**Comment by johnhooks** [comment]
@erikyo yes that looks perfect, but lets do it in a separate PR. Getting storybook functional is a very different issue than deploying to GitHub Pages.

We will also probably need @Sephsekla to enable using GitHub Actions to build GitHub Pages in the settings tab of the repo.

**Review by johnhooks — APPROVED** [review]
LGTM

File: .gitignore
Comment (erikyo) [reply .gitignore]: but you are right maybe a comment would be nice to separate things (for example, I hadn't noticed that he had a line underneath)

**Review by johnhooks — APPROVED** [review]
There were no changes since the last approval, only rebased onto `develop`. LGTM

**Review by johnhooks — DISMISSED** [review]

**Comment by Sephsekla** [comment]
@erikyo it looks like this might resolve a lot of the issues in https://github.com/WordPress/wp-feature-notifications/issues/241, anything I can do to unblock right now?

File: package.json
Comment (Sephsekla) [inline package.json:1]: Could you resolve the conflicts with the develop branch on this file?

Should this not also come with some updates to the lockfile?

File: .gitignore
Comment (Sephsekla) [reply .gitignore]: I've started to make some updates to this in https://github.com/WordPress/wp-feature-notifications/pull/240

**Comment by erikyo** [comment]
> @erikyo it looks like this might resolve a lot of the issues in #241, anything I can do to unblock right now?

here problems are caused by the fact that storybook uses a different version of react, and in this pr I am separating storybook from the main repo and updating it to version 7.

**Review by Sephsekla — DISMISSED** [review]

**Comment by Sephsekla** [comment]
Blocked by https://github.com/storybookjs/storybook/issues/21900 until Storybook releases an update.

**Comment by Sephsekla** [comment]
FYI https://github.com/storybookjs/storybook/pull/21950 - we should hopefully get a 7.0.3 release very soon.

**Comment by erikyo** [comment]
@mirka can you tell me what is wrong here? I saw that you deal (brilliantly) with this in the Gutenberg repo and I guess you could help me... 😓

~~in short happens to me that using the Gutenberg `Button` component throws me this error 
`Cannot read properties of null (reading 'useMemo')`. Is there a conflict with react native? how can i solve it?~~

☝️ solved

**Comment by Sephsekla** [comment]
@erikyo I can't claim to know a lot about storybook, so just my best guess. Does storybook's build config use webpack (and so maybe the root level webpack config)? Because if we're using the wp-scripts config then that will extract `@wordpress` packages, including React imports.

**Comment by erikyo** [comment]
> Because if we're using the wp-scripts config then that will extract `@wordpress` packages, including React imports.

yeah i've tried with [this](https://storybook.js.org/docs/react/builders/webpack#import-a-custom-webpack-configuration) to import the webpack configuration but doesn't work

**Comment by johnhooks** [comment]
I know nothing about Storybook, but have you looked in here? I believe this folder contains the Gutenberg configuration and build process for Storybook. https://github.com/WordPress/gutenberg/tree/bf809a6f841bc7172846687e2669122bcee300c4/storybook

**Comment by johnhooks** [comment]
Can we use NPM workspaces to share the `node_modules` folder of root? It would allow having multiple `package.json` files to split up dependencies, but would still install them all when `npm install` is run in the project root directory.

**Comment by erikyo** [comment]
> I know nothing about Storybook

me too, but is worth to add it into our project! :)

> have you looked in here...

Yep digged into [gutenberg repo](https://github.com/WordPress/gutenberg/tree/trunk/storybook), [woocommerce blocks](https://github.com/woocommerce/woocommerce-blocks), [yoast UI lib](https://github.com/Yoast/wordpress-seo/tree/trunk/packages/ui-library) but no one looks like our case... maybe was me, but i prefer to split the package.json into multiple and keep the root one clean. This because AFAIK npm doesn't install 2 times the same package if a package.json in a lower level has the same dependency (so in short we aren't going to install the same @wordpress - packages twice).

forgot to mention that this issue happened from the update to storybook 7 onwards, previously it worked with the same paths

**Comment by johnhooks** [comment]
> This because AFAIK npm doesn't install 2 times the same package if a package.json in a lower level has the same dependency (so in short we aren't going to install the same https://github.com/WordPress - packages twice).

I believe it does install them twice, unless workspaces are configured.

**Comment by erikyo** [comment]
> > This because AFAIK npm doesn't install 2 times the same package if a package.json in a lower level has the same dependency (so in short we aren't going to install the same https://github.com/WordPress - packages twice).
> 
> I believe it does install them twice, unless workspaces are configured.

ok! Not 100% sure but actually my `storybook/node_modules/@wordpress` looks like this
image

can you give a check too? I am using `npm 9.6.3`

**Comment by erikyo** [comment]
> can you give a check too? I am using npm 9.6.3

ah, you are right the duplicated packages are stored into `storybook/node_modules/@wordpress/wp-feature-notifications/node_modules/@wordpress`... so yes the "@wordpress" packages are duplicated which is not really ideal!

@johnhooks, do you say it is worth it to switch to pnpm?

**Comment by johnhooks** [comment]
> ok! Not 100% sure but actually my storybook/node_modules/@wordpress looks like this

@erikyo, yes with the addition of `@wordpress/wp-feature-notifications` as a dependency this is what I see too. Though before when the `@wordpress/*` packages from NPM were listed in both the root and storybook package.json files, I saw them as fully installed in both locations.

**Comment by johnhooks** [comment]
@erikyo Actually with `@wordpress/wp-feature-notifications` installed with `"../" I only see the build assets in the `node_modules/@wordpress/wp-feature-notifications` directory, it's not a link to the root project directory.

**Comment by erikyo** [comment]
@Sephsekla we need to merge this one after #252 in order to update this one with the new drawer before updating the whole repo. Please take care of [that one](https://github.com/WordPress/wp-feature-notifications/pull/252) first, both of these pr's are IMHO a big and nice advancement, sorry for the number of commits/files changed but since those are structural changes doing them in various PR's would have stalled development considerably

File: .storybook/wp-notify-theme.js
Comment (Sephsekla) [inline .storybook/wp-notify-theme.js:1]: Only just noticed this, but could we update the naming here to reference WP Feature Notifications instead of WP Notify? Similar to what @johnhooks has been doing in PHP.

**Review by Sephsekla — DISMISSED** [review]
Nice!

Added one suggestion around naming, it doesn't necessarily have to be a blocker at this stage though.

**Comment by Sephsekla** [comment]
Oops, hadn't noticed that this now has a couple of conflicts from the drawer update. As soon as those are resolved,re-request review and I can approve.

**Comment by erikyo** [comment]
> Oops, hadn't noticed that this now has a couple of conflicts from the drawer update. As soon as those are resolved,re-request review and I can approve.

We have [branched this PR](https://github.com/erikyo/wp-feature-notifications/commits/docs/storybook-wip) to keep things clean, we are working on... and we are close to get it fully working! In addition there are some improvements over the previous version, such as the default open drawer (cc @johnhooks) and some useful modules as the one that performs a a11y checking automatically for each component.

File: .storybook/wp-notify-theme.js
Comment (erikyo) [reply .storybook/wp-notify-theme.js:1]: Many of these are already updated in the storybook-wip branch, my only doubt was about [css custom props](https://github.com/erikyo/wp-feature-notifications/blob/docs/storybook-wip/src/styles/notify/vars.scss#L7-L14) because `wp-notification-feature` is really too long to be used on this purpose! @Sephsekla @johnhooks probably [ because of these cases the](https://github.com/erikyo/wp-feature-notifications/blob/5878b4afafdb6e9c2fc77ebb4c07a5ed4f33087f/src/styles/notify/hub/notice.scss#L13) "wp-notifications" prefix might work better? What do you think?

Comment (johnhooks) [reply .storybook/wp-notify-theme.js:1]: I support this `wp-feature-notifications` is too long of a prefix, or even adopting something that is shorter like the performance plugin did:

https://github.com/WordPress/performance/blob/f7a46205dd2818cf59786d8d0a0370a78e3704c4/load.php#L39

I don't think we want to change the rest api to... do we?

`/wp-json/wp-feature-notifications/v1/notifications`

I can't think of anything that rolls off the tounge as well as `perflab` though... `featnote` sounds awful.

**Comment by erikyo** [comment]
@johnhooks, also we might want to provide in this PR the workflow for publishing the storybook? An idea of what it needs to look like is [this one](https://github.com/erikyo/wp-feature-notifications/blob/docs/storybook-ci/.github/workflows/storybook.yml) (and I got it [here](https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/storybook-pages.yml))

**Comment by johnhooks** [comment]
Setup node and install deps. I think it would work if you move into the `storybook` directory first`. If not, at least copy the caching logic to speed up the npm install.

```yml
            - name: Setup Node.js and install dependencies
              uses: ./.github/setup-node
```

Scratch that one above, I didn't realized that second one you linked is the Gutenberg repo. There are some changes I would suggest, can you add it to this PR and I'll make some inline comments?

**Comment by erikyo** [comment]
@johnhooks since i am not sure, I would prefer if you could add the workflow part to this PR, thanks! (I believe you have all the permissions to do that, but if not tell me)

File: storybook/stories/assets/fake_api.json
Comment (johnhooks) [inline storybook/stories/assets/fake_api.json]: @erikyo do we need a full copy of this file, or can we adapt the changes to the one already in `includes/restapi/fake_api.json`?

**Review by johnhooks — CHANGES_REQUESTED** [review]

File: storybook/stories/assets/fake_api.json
Comment (erikyo) [reply storybook/stories/assets/fake_api.json]: I copied the original fake_api file (well, was moved before the merge) because in the future will be removed from `/includes/`. For the moment we can link to the "oringinal" and remove this one, but then we will have to remember to move it again when we have a working api

Comment (johnhooks) [reply storybook/stories/assets/fake_api.json]: @erikyo Do you just want to move it now? Into something like a `data` folder in the project root. That way it has a permanent home that both can use, and makes sense when the PHP code no longer requires it.

Comment (erikyo) [reply storybook/stories/assets/fake_api.json]: yep I want to move it, but not in this PR! For the moment I did as you suggested, deleted the copy and linked it to the "original" one.

**Review by johnhooks — APPROVED** [review]
LGTM 🤘🏻

**Comment by mirka** [comment]
> @mirka can you tell me what is wrong here?

Apologies for the late response as I was catching up from vacation! Glad you all got it sorted 👍

**Pull request merged**
